### PR TITLE
📖 Add more documentation about the KCP pre-terminate hook

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Follow-up to https://github.com/kubernetes-sigs/cluster-api/pull/11137

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: api/v1beta1/machine_types.go [hunk 1]
```diff
@@ -61,6 +61,16 @@ const (
 	// search each annotation for during the pre-terminate.delete lifecycle hook
 	// to pause reconciliation of deletion. These hooks will prevent removal of
 	// an instance from an infrastructure provider until all are removed.
+	//
+	// Notes for Machines managed by KCP (starting with Cluster API v1.8.2):
+	// * KCP adds its own pre-terminate hook on all Machines it controls. This is done to ensure it can later remove
+	//   the etcd member right before Machine termination (i.e. before InfraMachine deletion). If the etcd member would be
+	//   removed earlier, kubelet would start failing and the Node drain would not work.
```
Comment (fabriziopandini) [inline api/v1beta1/machine_types.go:68]:
```suggestion
	//   the etcd member right before Machine termination (i.e. before InfraMachine deletion). 
```
this it is true only for K8s>=1.31, and it is already described it in in the next point.

Same in the migration page

**Review by fabriziopandini — COMMENTED** [review]
Thanks for documenting this!
only one nit

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
Fixed, automated cherry-pick won't work. v1.8-v1.9 doc doesn't exist there

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**